components/concept/storage/src/main/java/mozilla/components/concept/storage/Storage.kt

```diff
@@ -16,5 +21,38 @@ interface Storage : Cancellable {
     /**
      * Runs internal database maintenance tasks
      */
-    suspend fun runMaintenance()
+    suspend fun runMaintenance(dbSizeLimit: UInt)
+}
+
+/**
+ * Builds and returns a [PeriodicWorkRequest] based on [PeriodicWorkRequestBuilder]
+ * with a given [StorageMaintenanceWorker].
+ *
+ * @param repeatInterval Repeat interval of the periodic work request.
+ * @param repeatIntervalTimeUnit Time unit for the repeat interval of the work request.
+ * @param tag Work request's tag.
+ * @param constraints A block that returns the [Constraints] for the work.
+ * @return [PeriodicWorkRequest].
+ * */
+inline fun <reified T : StorageMaintenanceWorker> Storage.periodicStorageWorkRequest(
```

Comment:
This looks great. Let's move them to the `browser-storage` component or another support component if you have a stronger preference.

The `concept` components are a group of APIs and structures that other components or consumers would want to use in order to make them work with custom implementations. We want to avoid implementation specifics here and also to avoid them from having implementation-specific dependencies too.

Say for example, an app doesn't like our implementation of `browser-toolbar` but wants/needs to use the same toolbar API with other components that connect together. The `concept-toolbar` contains _only_ the interfaces and data classes needed to implement any browser toolbar in order for it to work with the rest of our components.

You can now use, say the [`feature-awesomebar`](https://searchfox.org/mozilla-mobile/rev/5f548600686287ca343e87b4f3ee65dc94707a1d/android-components/components/feature/awesomebar/build.gradle#28), with your own implementation of a `Toolbar` because it doesn't depend on `browser-toolbar` nor does it pull in the `browser-toolbar` dependencies for you if you just need those APIs/interfaces.